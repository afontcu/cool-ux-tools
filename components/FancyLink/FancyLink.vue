<template>
  <a :href="href" :title="title" class="FancyLink">
    <slot>{{ title }}</slot>
  </a>
</template>

<script>
export default {
  name: 'FancyLink',
  props: {
    href: { required: true, type: String },
    title: { required: true, type: String },
  },
  data () {
    return {}
  },
}
</script>

<style lang="scss">
.FancyLink {
  position: relative;
  transition: color $global-transition ease;

  &:hover {
    color: #aaa;

    &::before {
      width: 100%;
      left: 0;
    }

  }

  &::before {
    content: "";
    position: absolute;
    top: calc(100% + 1px);
    width: 0;
    right: 0;
    height: 3px;
  }

  &::before {
    transition: width 0.3s cubic-bezier(0.53, 0.02, 0.52, 0.99) 0.1s;
    background: map-get($global-colors, 'branding-primary');
  }
}
</style>
