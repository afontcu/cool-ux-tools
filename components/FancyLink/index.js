import FancyLink from './FancyLink'

export default FancyLink
