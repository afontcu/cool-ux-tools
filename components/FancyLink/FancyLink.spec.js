import FancyLink from './index.js'
import { mount } from '@vue/test-utils'

const propsData = {
  href: 'href.com',
  title: 'someTitle',
}

describe('FancyLink.vue', () => {
  it('is a component', () => {
    const wrapper = mount(FancyLink, { propsData })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders main node', () => {
    const wrapper = mount(FancyLink, { propsData })
    expect(wrapper.classes()).toContain('FancyLink')
  })

  it('renders a snapshot', () => {
    const wrapper = mount(FancyLink, { propsData })
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
