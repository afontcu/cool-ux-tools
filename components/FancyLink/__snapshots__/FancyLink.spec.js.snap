// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyLink.vue renders a snapshot: default 1`] = `

<a href="href.com"
   title="someTitle"
   class="FancyLink"
>
  someTitle
</a>

`;
